gammapy/utils/coordinates/fov.py

```diff
@@ -1,122 +1,192 @@
+    target_fov = SkyCoord(-lon, lat, frame=fov_frame)
+
+    # Transform into celestial system (need not be ICRS)
+    target_sky = target_fov.icrs
+
+    return target_sky.ra, target_sky.dec
 
 
 def sky_to_fov(lon, lat, lon_pnt, lat_pnt):
     """Make a transformation from sky coordinates to field-of-view coordinates.
 
     Parameters
     ----------
-    lon : array_like
-        Sky longitude coordinate to be transformed, in degrees
-    lat : array_like
-        Sky latitude coordinate to be transformed, in degrees
-    lon_pnt : array_like
-        Longitude coordinate of the pointing position, in degrees
-    lat_pnt : array_like
-        Latitude coordinate of the pointing position, in degrees
+    lon : `~astropy.units.Quantity`
+        Sky longitude coordinate to be transformed
+    lat : `~astropy.units.Quantity`
+        Sky latitude coordinate to be transformed
+    lon_pnt : `~astropy.units.Quantity`
+        Longitude coordinate of the pointing position
+    lat_pnt : `~astropy.units.Quantity`
+        Latitude coordinate of the pointing position
 
     Returns
     -------
-    lon_t : array_like
-        Field-of-view longitude coordinate, in degrees
-    lat_t : array_like
-        Field-of-view latitude coordinate, in degrees
+    lon_t : `~astropy.units.Quantity`
+        Field-of-view longitude coordinate
+    lat_t : `~astropy.units.Quantity`
+        Field-of-view latitude coordinate
     """
-    lon_ = -lon
-
-    # compute cartesian coordinates
-    x = np.cos(np.deg2rad(lat)) * np.cos(np.deg2rad(lon_))
-    y = np.cos(np.deg2rad(lat)) * np.sin(np.deg2rad(lon_))
-    z = np.sin(np.deg2rad(lat))
-
-    # make sure vector is properly normalised
-    assert np.allclose(x ** 2 + y ** 2 + z ** 2, 1)
-
-    # transform
-    lat_pnt = np.deg2rad(90 - lat_pnt)
-    lon_pnt = -np.deg2rad(lon_pnt)
-    x_t = (
-        x * np.cos(lat_pnt) * np.cos(lon_pnt)
-        + y * np.sin(lon_pnt) * np.cos(lat_pnt)
-        - z * np.sin(lat_pnt)
-    )
-    y_t = -x * np.sin(lon_pnt) + y * np.cos(lon_pnt)
-    z_t = (
-        x * np.sin(lat_pnt) * np.cos(lon_pnt)
-        + y * np.sin(lat_pnt) * np.sin(lon_pnt)
-        + z * np.cos(lat_pnt)
-    )
-
-    # switch coordinates due to axis convention
-    x_ = z_t
-    y_ = y_t
-    z_ = -x_t
-
-    # compute new lon, lat
-    lon_t = np.rad2deg(np.arctan2(y_, x_))
-    lat_t = np.rad2deg(np.arcsin(z_))
-
-    return lon_t, lat_t
+
+    # Create a frame that is centered on the pointing position
+    center = SkyCoord(lon_pnt, lat_pnt)
+    fov_frame = SkyOffsetFrame(origin=center)
+
+    # Define coordinate to be transformed.
+    target_sky = SkyCoord(lon, lat)
+
+    # Transform into FoV-system
+    target_fov = target_sky.transform_to(fov_frame)
+
+    # Switch sign of longitude angle since this axis is
+    # reversed in our definition of the FoV-system
+    return -target_fov.lon, target_fov.lat
+
+
+# def fov_to_sky(lon, lat, lon_pnt, lat_pnt):
```

Comment:
We never keep commented out code around. Can be looked up in the git history. Please delete it.